### ISSUE TITLE
[bench request]: Add decider-35b-a3b (Mapika/decider-35b-a3b, TypeSafe wire format)

### ISSUE THREAD

**Issue by Mapika** [fstandhartinger/jevbench#4]
Request to add [decider-35b-a3b](https://huggingface.co/Mapika/decider-35b-a3b) (Qwen3.5-35B-A3B-Base, 34.7B parameters with 3B active per token, Apache-2.0 weights and code) to the ranked systems. It is the larger sibling of decider-2b (#2); both serve the same wire format, so one adapter covers both.

**What it is.** A one-pass decision model: the letter logits at an answer slot are read for every option, no decoding. The supervised recipe of decider-2b (one epoch of cross-entropy on the slot readout over about 95 public decision datasets and Jev's input shapes) applied to the 35B mixture-of-experts base with the routed experts frozen and the Muon optimizer; no RL stage. Every question is scored in its own row, so option order and the other questions do not change an answer. Code and recipe: https://github.com/Mapika/decider (`moe/`, `docs/HISTORY.md`).

It serves TypeSafe's `POST /v1/systemone`, so your `typesafe` adapter applies unchanged:

```sh
git clone https://github.com/Mapika/decider && cd decider
uv venv --python 3.12 .venv312 && uv pip install -p .venv312/bin/python -e ".[serve]"
scripts/serve.sh Mapika/decider-35b-a3b 8000      # one CUDA GPU with 80 GB or more, bf16, 65 GB of weights; downloads on first use
                                                  # transformers>=5.17 and torch>=2.14 (grouped-GEMM experts), flash-linear-attention

python -m jevbench.cli run --tasks datasets/public/original.jsonl \
  --adapter typesafe --endpoint http://127.0.0.1:8000 --key-env '' --model decider-35b-a3b \
  --cost-basis self_hosted_gpu --reserve-usd 0 ...
```

`noul` returns `noul`, `choice` and `score` return `probabilities` keyed by option name and level index, as in your adapter's mapping. An NVFP4 build for vLLM and TensorRT-LLM on Blackwell is at [decider-35b-a3b-nvfp4](https://huggingface.co/Mapika/decider-35b-a3b-nvfp4) (19.6 GB; 1.0 to 1.5 accuracy points under bf16 on our fixtures); its card has the readout through vLLM, but the HTTP server above runs the bf16 weights.

**Our own numbers on the public items**, with the request your adapter builds, argmax over the exact label set, one request at a time:

| tier | items | decider-35b-a3b v1 | decider-2b v10 |
|---|---|---|---|
| easy | 48 | 1.000 | 1.000 |
| standard (original) | 72 | 0.972 | 0.847 |
| hard | 111 | 0.676 | 0.459 |

Hard-tier top-label ECE is 0.15. The remaining hard misses are temporal-numeric items (5 of 15 correct), long-policy and judge-hard items; adversarial, trap and hard-routing items are all correct. Standard-tier misses are one adequacy and one policy item.

**Speed and cost basis.** One NVIDIA B300, eager PyTorch (the CUDA-graph engine of the package is not yet enabled for this architecture): 47 ms per request with three questions on a 90-token state, about 520 decisions per second in batches of 64. decider-2b on the same machine serves at 4 ms per request. If you rank one of the two, this is the stronger entry; the 2B is the cheaper one.

**Comment by fstandhartinger** [comment]
Thanks — this is logged for reachability checks and the JevBench measurement queue. We’ll only make benchmark claims after our own run.

- Harold

**Comment by fstandhartinger** [comment]
decider-35b-a3b ran the full frozen v1.2 set — 534 decisions including the held-out ones — through your server and
FP8 weights on our RunPod H100 NVL, one request at a time (`DECIDER_COMPILE=0`, lazy shape capture).

**#10 of 36, JevBench Score 68.9** (Intelligence 86.3, Calibration 71.5, Speed 80.8, Cost 45.3). Accuracy: easy 100.0 %, standard 96.9 %, judge 91.1 %, hard 65.5 %. Latency p50 0.29 s raw, 0.73 s after the standard self-host adjustment (x2 + 0.15 s); our pod was in Canada, so the raw figure includes the trip from our server in Germany; cost $0.0666 per 1,000 decisions.

Page: https://benchmarkheaven.com/jev-models · data and method: https://github.com/fstandhartinger/jevbench/tree/v1.2.8

- Harold

**Comment by Mapika** [comment]
Thanks for running it, and for publishing the run conditions before the run.

One clarification on the footnote and the comment above ("published FP8 weights"). The weights on the Hub at revision `30c5f98` are bf16 (65 GB). The FP8 comes from our server: `decider.serve` defaults to `DECIDER_FP8=1`, which converts the 110 large dense linear layers to e4m3 at load; the routed experts stay bf16. So "FP8" describes your run correctly and "published FP8 weights" does not. Suggested wording: "bf16 weights; the author's server converts the large linear layers to FP8 at load (its default)".

Our model card lists that path as untested on this architecture, so we measured it: the 231 public items, the same start-up as yours (`DECIDER_COMPILE=0`), one B300, requests built the way your `typesafe` adapter builds them, one at a time.

| server setting | easy (48) | standard (72) | hard (111) | hard top-label ECE | hard Brier |
|---|---|---|---|---|---|
| default (`DECIDER_FP8=1`), as in your run | 1.000 | 0.972 | 0.667 (74) | 0.172 | 0.469 |
| `DECIDER_FP8=0` (bf16) | 1.000 | 0.972 | 0.676 (75) | 0.157 | 0.449 |

The argmax differs on 5 of the 231 items, all in the hard tier (3 lost and 2 gained under FP8); the mean total variation between the two answer distributions is 0.018. That is one hard item and about 0.015 of ECE. It does not explain the row and we are not asking for a re-run. If you re-measure for another reason, `DECIDER_FP8=0` is the configuration our card's numbers come from.

The calibration gap on the hard tier is ours to fix: a single temperature cannot close it without hurting the easier tiers, so it needs work on the model, which is under way.

**Comment by fstandhartinger** [comment]
Congratulations on the JevBench v1.4 release, Mapika! Thanks for building and sharing your system.

- decider-35b-a3b: #14 at 41.18
- decider-2b: #30 at 30.74

decider-35b-a3b was #8 in v1.3; decider-2b was #23 in v1.3. V1.4 blends in 308 fresh sealed decisions, uses an equal-weight harmonic mean, and applies separate Speed and Cost gates. For your leading row, sealed accuracy was 31.5% versus 83.1% on the public slice. The >25-point gap adjustment applied; the below-50 Intelligence/Cost gates also applied.

Full board and method: https://benchmarkheaven.com/jev-models
